Rhinoceros_Engine/Compute/CollectAllModelData.cs

```diff
@@ -0,0 +1,112 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using System.IO;
+using BH.oM.Reflection;
+using System.Drawing;
+using Rhino.Geometry;
+
+namespace BH.Engine.Rhinoceros
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Opens a series of rhino models and fetches all the data from them.")]
+        [Input("fileNames", "Full path filenames to the files to be extracted.")]
+        [Input("go", "Toggles on and off the execution of the method.")]
+        [MultiOutput(0, "layerName", "The name of the model and the layer name from the opened model separated with '::'. Can be used with the Human grasshopper plugin to generate new layers.")]
+        [MultiOutput(1, "colour", "The colour of the layer.")]
+        [MultiOutput(2, "geometry", "The geometry from the models.")]
+        public static Output<List<string>, List<Color>, List<GeometryBase>> CollectAllModelData(List<string> fileNames, bool go = false)
+        {
+            if (!go)
+                return new Output<List<string>, List<Color>, List<GeometryBase>>();
+
+            //bool to stop the method from triggering multiple times at once
+            //Without this, the execution when called from GH end up calling the extraction and opening of the models multiple times in a strange fashion
+            //And the output gets duplicated and screwed up.
+            //TODO: find a better solution for this
+            if (m_isCollecting)
+                return new Output<List<string>, List<Color>, List<GeometryBase>>();
+            else
+                m_isCollecting = true;
+
+            try
+            {
+
+                List<string> layerNames = new List<string>();
+                List<Color> colours = new List<Color>();
+                List<GeometryBase> geometries = new List<GeometryBase>();
+
+                foreach (string fileName in fileNames)
+                {
+                    string name = Path.GetFileName(fileName);
+                    Rhino.RhinoDoc.OpenFile(fileName);
+
```

Comment:
I would try `using Rhino.FileIO` and then read 3dm files without actually opening them - I think this will remove some of the issues noted with the component triggering on actually opening the file.
```suggestion
File3dm file3Dm = File3dm.Read(fileName);
```

Reply:
Omg, this is so much cleaner/better working. Thanks! Will update!

Reply:
I just pushed a commit. Hope that is ok? Delete it if you don't want me meddling - just had a few mins to try it out.